Sources/Commands/Options.swift

```diff
@@ -9,22 +9,106 @@
 */
 
 import Basic
-import POSIX
-import PackageModel
 import Utility
 
-public class Options {
-    public var chdir: AbsolutePath?
-    public var enableNewResolver = false
-    public var colorMode: ColorWrap.Mode = .Auto
-    public var verbosity: Int = 0
-    public var buildPath: AbsolutePath?
+public class ToolOptions {
+    /// The custom working directory that the tool should operate in.
+    public var chdir: AbsolutePath? {
+        guard let path = result.get(chdirOption) else { return nil }
+        return AbsolutePath(path, relativeTo: currentWorkingDirectory)
+    }
+    private let chdirOption: OptionArgument<String>
 
-    public init()
-    {}
-}
+    /// The custom build directory, if provided.
+    public var buildPath: AbsolutePath? {
+        return pathRelativeToWorkingDir(buildPathOption)
+    }
+    private let buildPathOption: OptionArgument<String>
```

Comment:
I wonder if we should try and come up with better infrastructure for this pattern? Separating the options and values is good for the lowest level implementation, but I think a convenient facility for deriving simple structs from options would be good to also have in the infrastructure not in each tool.